apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx

```diff
@@ -152,6 +144,20 @@ export default function RubricsContainer({
     setLearningGoalList(newLearningGoalData);
   };
 
+  const deleteLearningGoal = idToDelete => {
+    const newLearningGoalData = learningGoalList.map(learningGoal => {
+      if (idToDelete === learningGoal.id) {
+        return {
+          ...learningGoal,
+          _destroy: true,
```

Comment:
This was the piece of magic needed.  This is what allows `nested_attributes` to be deleted.

Reply:
I do love a piece of Rails magic forced into the React universe. 🙃 Looks good.

I'm curious and trying to learn some React intuition: is it important that the resulting array that gets sent to the `setLearningGoalList` state be "new"? I can imagine writing something like:

```
const newLearningGoalData = learningGoalList.map(learningGoal => {
  if (idToDelete == learningGoal.id) {
    learningGoal._destroy = true;
  }
  return learningGoal;
});
```

In which case, since technically the resulting list from the map is identical to the original list:

```
learningGoalList.forEach(learningGoal => {
  if (idToDelete == learningGoal.id) {
    learningGoal._destroy = true;
  }
});
```

But then the list would be "The Same" as the list already in the state. (We only altered an item within the list.) React would have to know that an item within the array was updated instead? Is that a normal thing?

Anyway, constructing a new list with the items from the old list is also fine.

Reply:
Great question!  I had read somewhere that you could not directly update state and had to use setState instead.

I went back to confirm that this was true, which was nice because it allowed me to re-read the documentation now that I know more than when I first read it too.  Here is the [documentation](https://react.dev/reference/react/useState#updating-objects-and-arrays-in-state) which also links to updating arrays and objects in state too.   Since this component has an array of objects, I think the logic still holds, that what is stored in state should be treated as read-only. 

That being said, I tried replacing the function without using setState and it seemed to work and didn't give me any errors.  I am not sure where that would be caught if it was a problem (maybe in review??). 

When I started investigating why you shouldn't mutate state directly, it got fairly technical fairly fast... the gist of it was "it can cause problems".